### PR TITLE
fix wrong metrics for flushing segments

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #24294

### CODE REVIEW

**Comment by MrPresent-Han** [comment]
rerun ut

File: internal/datacoord/meta.go [hunk 1]
```diff
@@ -1022,10 +1022,19 @@ func (m *meta) PrepareCompleteCompactionMutation(compactionLogs []*datapb.Compac
 	}
 	segment := NewSegmentInfo(segmentInfo)
 	metricMutation.addNewSeg(segment.GetState(), segment.GetNumOfRows())
+	if segment.GetNumOfRows() == 0 {
+		metricMutation.append(segment.GetState(), commonpb.SegmentState_Dropped, segment.GetNumOfRows())
+		segment.State = commonpb.SegmentState_Dropped
+		log.Info("compaction generated segment is empty, set to dropped state by default",
+			zap.Int64("collection ID", segment.GetCollectionID()),
+			zap.Int64("partition ID", segment.GetPartitionID()),
```
Comment (congqixia) [inline internal/datacoord/meta.go:1030]: Shall we keep this addNewSeg mutation even the segment row count is zero?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**